hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java

```diff
@@ -232,10 +241,26 @@ public JobConfig setSnapshotIntervalMillis(long snapshotInterval) {
     @Nonnull
     @SuppressWarnings("rawtypes")
     public JobConfig addClass(@Nonnull Class... classes) {
-        checkNotNull(classes, "Classes can not be null");
+        checkNotNull(classes, "Classes cannot be null");
         for (Class<?> clazz : classes) {
             ResourceConfig cfg = new ResourceConfig(clazz);
             resourceConfigs.put(cfg.getId(), cfg);
+
+            ClassLoader classLoader = clazz.getClassLoader();
+            String packagePrefixPath = Optional.ofNullable(clazz.getPackage())
+                                               .map(pakage -> pakage.getName().replace('.', '/') + "/")
+                                               .orElse("");
+            Pattern nestedPattern = Pattern.compile(clazz.getName()
+                                                         .replace(".", "\\/")
+                                                         .replace("$", "\\$") + "\\$.+" + "\\.class");
```

Comment:
You can simplify this by first building a literal class string (no escaping) and then calling `Pattern.quote()` on it.